src/features/Login/hooks/useLogin.ts

```diff
@@ -0,0 +1,61 @@
+import {useState} from 'react';
+import {FirebaseAuthTypes} from '@react-native-firebase/auth';
+import AuthAPI from '../../../services/API/Auth';
+import {validateEmail, validatePassword} from '../../../utils/validators';
+
+const useLoginForm = () => {
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const [isLoading, setIsLoading] = useState(false);
+
+  const onNicknameChange = (text: string) => {
+    setError('');
+    setEmail(text);
+  };
+
+  const onPasswordChange = (text: string) => {
+    setError('');
+    setPassword(text);
+  };
+
+  const loginHandler = async () => {
+    try {
+      setIsLoading(true);
+      const validationError =
+        validateEmail(email) || validatePassword(password);
+      if (validationError) {
+        setError(validationError);
+        return;
+      }
+
+      await AuthAPI.login({email, password});
+    } catch (e) {
+      const error = e as FirebaseAuthTypes.NativeFirebaseAuthError;
+      if (
+        error.code === 'auth/invalid-email' ||
+        error.code === 'auth/wrong-password'
```

Comment:
this looks a bit cleaner like:
['auth/invalid-email',  'auth/wrong-password'].includes(error.code)